### DISCUSSION TITLE
Hosting files jbrowse2

### DISCUSSION THREAD

**Discussion by Vishvak2000** [GMOD/jbrowse-components#2216] [Q&A]
I have hosted all my assembly/track files on a publicly available s3 bucket.  The urls are in the format https://<bucket-name>.s3.us-west-1.amazonaws.com/Sequences/Core/Core_epitopes.sorted.gff.gz . When I put the s3 urls into my browser, it downloads the file so I do think the file has the correct access restrictions. However, when I try to use these s3 bucket links add tracks to JBrowse2, I get an "Failed to fetch" error.  Are the s3 urls in the correct format? If not are there any alternatives as to where I could host my files?

**Comment by garrettjstevens** [comment]
We use S3 for some hosting internally and it works, so I think you should be able to get S3 to work for your needs. Do you have the correct [CORS](https://web.dev/cross-origin-resource-sharing/) configuration on your S3 bucket? Having the wrong CORS configuration could stop JBrowse from working, even if the files have the right URL. An S3 CORS configuration with the permissions JBrowse needs might look something like:

```json
[
    {
        "AllowedHeaders": [
            "*"
        ],
        "AllowedMethods": [
            "GET",
            "HEAD"
        ],
        "AllowedOrigins": [
            "*"
        ],
        "ExposeHeaders": [
            "Accept-Ranges",
            "Content-Range",
            "Content-Encoding",
            "Content-Length"
        ],
        "MaxAgeSeconds": 3000
    }
]
```
